MvvmCross/ViewModels/MvxAppStart.cs

```diff
@@ -3,32 +3,47 @@
 // See the LICENSE file in the project root for more information.
 
 using System.Threading;
-using System.Threading.Tasks;
 using MvvmCross.Exceptions;
 using MvvmCross.Logging;
 using MvvmCross.Navigation;
 
 namespace MvvmCross.ViewModels
 {
-    public class MvxAppStart<TViewModel>
-        : IMvxAppStart
-        where TViewModel : IMvxViewModel
+    public abstract class MvxAppStart : IMvxAppStart
     {
-        protected readonly IMvxNavigationService NavigationService;
```

Comment:
@nmilcoff Love this refactor. Is there a reason why the navigation service can't be pushed down to the non-generic version?

Reply:
Thanks! Yeah, so the problem is that there isn't enough information to use it... the generic version adds the ViewModel the app wants to navigate to. 
The non generic version will allow apps to take the decision without compromising stability.

Reply:
But won't the MvxAppStart be kinda useless if they don't have access to the NavigationService. Instead of having developers write Mvx.Resolve in order to get it, we might as well include it in the non-generic version so it can be injected into the constructor. We can still leave the Startupup method as abstract, so it shouldn't affect stability

Reply:
That should be up to the developer, who can easily inject the NavigationService on its own implementation. I don't see a reason to include a dependency on a class that won't be used.

Please look at my implementation for the [StarWarsSample](https://github.com/MvvmCross/MvvmCross-Samples/pull/114/files#diff-56bd2bea371da509b91bfb9c38d72189R7). Because there's no "base" class to inherit from, I had to include some logic for `IsStarted` by myself.